### PR TITLE
PP-10875 Remove request library usage from tests

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Replaced request library usage in tests with Axios. This is to eventually remove the request library dependency which is not maintained anymore

### CODE REVIEW

**Review by SandorArpa — DISMISSED** [review]
LOKTM

File: test/cypress/plugins/index.js [hunk 1]
```diff
@@ -33,27 +33,35 @@ module.exports = (on, config) => {
...
-          protocol: 'http',
-          defaultResponse: {
-            statusCode: 404,
-            body: 'No stub predicate matches the request',
-            headers: {}
-          },
-          stubs
-        }
+      return new Promise((resolve, reject) => {
+        axios.post(mountebankImpostersUrl,
```
Comment (iqbalgds) [inline test/cypress/plugins/index.js:37]: Is it possible to change this to await/async?
As we prefer using that syntax as per our Node style guide:
https://gds-way.cloudapps.digital/manuals/programming-languages/nodejs/#use-of-async-await

**Review by iqbalgds — CHANGES_REQUESTED** [review]
Looks good - left a comment

File: test/cypress/plugins/index.js (hunk 1 above)
Comment (kbottla) [reply test/cypress/plugins/index.js:37]: Returning promise is fine in this case as this is expected by Cypress https://github.com/cypress-io/cypress/pull/6920/files

Comment (iqbalgds) [reply test/cypress/plugins/index.js:37]: Not sure if this is possible - but can we return a promise but change the axios code to use Async/Await?

E.g.

return new Promise(async (resolve, reject) => {
  try {
    // Axios code
    resolve()
  }catch (error){
    reject()
  }
})

Comment (kbottla) [reply test/cypress/plugins/index.js]: updated

Comment (kbottla) [reply test/cypress/plugins/index.js:37]: actually, it is not recommended to use async when returning promise https://eslint.org/docs/latest/rules/no-async-promise-executor

Comment (kbottla) [reply test/cypress/plugins/index.js]: I have removed returning Promise but it is not possible to change Axios usage to async/await without changing setupStubs to async

Comment (iqbalgds) [reply test/cypress/plugins/index.js]: Thanks for looking into that - I will approve

**Review by iqbalgds — APPROVED** [review]

**Pull request merged**